slices/playerLocationSlice.ts

```diff
@@ -0,0 +1,42 @@
+import { createSlice } from "@reduxjs/toolkit";
+import type { PayloadAction } from "@reduxjs/toolkit";
+import type { RootState } from "../store";
+
+type PlayerLocation = {
+  x: number;
+  y: number;
+};
+
+const initialState: PlayerLocation = {
+  x: 3,
+  y: 3,
+};
+
+export const playerLocationSlice = createSlice({
+  name: "playerLocation",
+  initialState,
+  reducers: {
+    up: (state) => {
+      state.y += 1;
```

Comment:
Super cool how we can just mutate state right here even though this would normally be bad practice. Immer is great!